hack/bin/integration-test.sh

```diff
@@ -1,9 +1,81 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 NAMESPACE=${NAMESPACE:-test-integration}
+# set to 1 to disable running helm tests in parallel
+HELM_PARALLELISM=${HELM_PARALLELISM:-1}
 
-echo "Running integration tests on wire-server"
+echo "Running integration tests on wire-server with parallelism=${HELM_PARALLELISM} ..."
 
 CHART=wire-server
-helm test --logs -n "${NAMESPACE}" "${NAMESPACE}-${CHART}" --timeout 900s
+tests=(cargohold gundeck federator spar brig galley)
+
+summary() {
+    echo "==============="
+    echo "=== summary ==="
+    echo "==============="
+    printf '%s\n' "${tests[@]}" | parallel echo "=== tail {}: ===" ';' tail -2 logs-{}
+
+    for t in "${tests[@]}"; do
+        x=$(cat "stat-$t")
+        if ((x > 0)); then
+            echo "$t-integration FAILED ❌. pfff..."
+        else
+            echo "$t-integration passed ✅."
+        fi
+    done
+}
+
+# Run tests in parallel using GNU parallel (see https://www.gnu.org/software/parallel/)
+# The below commands are a little convoluted, but we wish to:
+# - run integration tests. If they fail, keep track of this, but still go and get logs, so we see what failed
+# - run all tests. Perhaps multiple flaky tests in multiple services exist, if so, we wish to see all problems
+mkdir -p ~/.parallel && touch ~/.parallel/will-cite
```

Comment:
Will this directory need to be cleaned up to run the test next time?

Reply:
this is just a marker file to avoid gnu parallel complaining about wanting to be cited in academic papers. No need for cleanup if this runs locally.

(the logs-brig and stat-brig files so far remain on disk if you run the integration test locally)

Reply:
(update: the script will now clean up these local temporary files by default)